### PR TITLE
feat(executor): Add user agent to workflow executor

### PR DESCRIPTION
**Pull request by terrytangyuan**
We have many different applications calling apiserver and setting the user agent would allow us to distinguish the API calls  among those applications for further traffic control/monitoring. This would also prepare us to try different executors in https://github.com/argoproj/argo/pull/4998 and monitor their relevant traffic.

Signed-off-by: terrytangyuan <terrytangyuan@gmail.com>

Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

### CODE REVIEW

File: cmd/argoexec/commands/root.go [hunk 1]
```diff
@@ -72,6 +74,7 @@ func NewRootCommand() *cobra.Command {
 func initExecutor() *executor.WorkflowExecutor {
 	log.WithField("version", argo.GetVersion().Version).Info("Starting Workflow Executor")
 	config, err := clientConfig.ClientConfig()
+	config = restclient.AddUserAgent(config, fmt.Sprintf("argo-workflow-%s-executor", os.Getenv(common.EnvVarContainerRuntimeExecutor)))
```
Comment (alexec) [inline cmd/argoexec/commands/root.go:77]: user agent should probably be something like
```
User-Agent: <product> / <product-version> <comment>
```

E.g.

```
argo-workflows/v3.0.0 executor/docker
```

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent

Comment (terrytangyuan) [reply cmd/argoexec/commands/root.go]: Thanks! Just updated the UA to follow the convention.

File: cmd/argoexec/commands/root.go [hunk 2]
```diff
@@ -72,9 +72,11 @@ func NewRootCommand() *cobra.Command {
 }
 
 func initExecutor() *executor.WorkflowExecutor {
-	log.WithField("version", argo.GetVersion().Version).Info("Starting Workflow Executor")
+	version := argo.GetVersion()
+	log.WithField("version", version).Info("Starting Workflow Executor")
 	config, err := clientConfig.ClientConfig()
-	config = restclient.AddUserAgent(config, fmt.Sprintf("argo-workflow-%s-executor", os.Getenv(common.EnvVarContainerRuntimeExecutor)))
+	executorType := os.Getenv(common.EnvVarContainerRuntimeExecutor)
+	config = restclient.AddUserAgent(config, fmt.Sprintf("argo-workflows/%s executor/%s", version.Version, executorType))
```
Comment (alexec) [inline cmd/argoexec/commands/root.go:79]: @jessesuen thoughts on this User-Agent?

**Comment by alexec** [comment]
Awaiting feedback from @jessesuen.

**Review by jessesuen — APPROVED** [review]

**Comment by jessesuen** [comment]
Seems like it would be beneficial without any risk.

**Pull request merged**